firefox-ios/Ecosia/Core/Environment/Environment.swift

```diff
@@ -7,17 +7,85 @@ import Foundation
 public enum Environment: Equatable {
     case production
     case staging
+    case debug
 }
 
 extension Environment {
 
     public static var current: Environment {
+        /*
+         * Why not MOZ_CHANNEL_BETA compilation flags?
+         * - Project configs had SWIFT_ACTIVE_COMPILATION_CONDITIONS = ""; blocking xcconfig inheritance
+         * - Multiple BetaDebug configs with same name, Xcode uses wrong one
+         * - EcosiaTesting.xcconfig works because it sets explicit value, not empty string
+         * 
+         * Solution: Bundle ID detection is more reliable than build config inheritance
+         */
+
+        // Determine environment: staging has "firefox" in bundle ID, others use compilation flags
+        guard let bundleId = Bundle.main.bundleIdentifier else { return .production }
+
+        // Staging always has "firefox" in bundle ID (e.g., com.ecosia.ecosiaapp.firefox)
+        if bundleId.contains("firefox") {
+            return .staging
+        }
+
+        // For production vs debug (both use com.ecosia.ecosiaapp), use compilation flags
         #if MOZ_CHANNEL_RELEASE
         return .production
         #else
-        return .staging
+        return .debug
         #endif
     }
+
+    // Alternative: Bundle ID based detection
+    public static var currentFromBundleId: Environment {
+        guard let bundleId = Bundle.main.bundleIdentifier else {
+            return .debug
+        }
+
+        switch bundleId {
+        case "com.ecosia.ecosiaapp":
+            return .production
+        case "com.ecosia.ecosiaapp.firefox":
+            return .staging
+        default:
+            return .debug
+        }
+    }
+
+    // Alternative: Environment variable based detection
+    public static var currentFromEnvironmentVariable: Environment {
+        if let envVar = ProcessInfo.processInfo.environment["ECOSIA_ENVIRONMENT"] {
+            switch envVar.lowercased() {
+            case "production":
+                return .production
+            case "staging", "beta":
+                return .staging
+            default:
+                return .debug
+            }
+        }
+        return .debug
+    }
+
+    // Alternative: Info.plist based detection
+    public static var currentFromInfoPlist: Environment {
+        guard let path = Bundle.main.path(forResource: "Info", ofType: "plist"),
+              let plist = NSDictionary(contentsOfFile: path),
+              let envString = plist["EcosiaEnvironment"] as? String else {
+            return .debug
+        }
+
+        switch envString.lowercased() {
+        case "production":
+            return .production
+        case "staging", "beta":
+            return .staging
+        default:
+            return .debug
+        }
+    }
```

Comment:
I guess this alternatives are here just for showing? Would be nice to remove them before merging.

But to be honest, I prefer these approaches rather than mixing two different things together like currently 😅 Any reason for not using the bundle check or `ECOSIA_ENVIRONMENT`?

Reply:
Yes they are 👍 .
Should have put a more explanatory comment.